packages/api/cms-api/src/common/filter/mikro-orm.ts

```diff
@@ -150,12 +150,21 @@ export function filtersToMikroOrmQuery(
     return genericFilter(filter);
 }
 
+const splitSearchString = (search: string) => {
+    const regex = /(["'])(?:(?=(\\?))\2.)*?\1|\S+/g;
+    const matches = search.match(regex) || [];
+
+    return matches.map((match) => {
+        const unescaped = match.replace(/\\(['"])/g, "$1");
```

Comment:
Could you please explain what these lines do? I can't understand it simply by reading it.

Reply:
Sure.

line 158: remove escape characters from a matched string => eg. "\"quoted\"" 
line 159: checks if a match is quoted with ' or "
line 160: if it is quoted it extracts the actual string from the quote by "slicing away" the quotes otherwise it'll use just the unescaped match
line 161: format the string as needed (I don't know why its needed that way but its the same in the old version)

Should I comment the file as well?

Reply:
- [ ] add comment to regex